modules/centrifuge/kreport/main.nf

```diff
@@ -0,0 +1,37 @@
+process CENTRIFUGE_KREPORT {
+    tag "$meta.id"
+    label 'process_low'
+
+    // TODO nf-core: List required Conda package(s).
+    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
+    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
+    conda (params.enable_conda ? "bioconda::centrifuge=1.0.4_beta" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/centrifuge:1.0.4_beta--h9a82719_6':
+        'quay.io/biocontainers/centrifuge:1.0.4_beta--h9a82719_6' }"
+
+    input:
+    tuple val(meta), path(results)
+    path db
+
+    output:
+    tuple val(meta), path('*kreport.txt')                , emit: kreport
+    path "versions.yml"                                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    db_name=`find -L ${db} -name "*.1.cf" -not -name "._*"  | sed 's/.1.cf//'`
+    centrifuge-kreport -x \$db_name ${prefix}.results.txt > ${prefix}.kreport.txt
```

Comment:
```suggestion
    centrifuge-kreport -x \$db_name {results}.txt > ${results}.txt
```

Also add warning as in point 5 in the guidelines [^1]

[^1]: https://nf-co.re/docs/contributing/modules#naming-conventions

Reply:
Wouldn't we need the `prefix` here?

The output of centrifuge_centrifuge is: ` -S ${prefix}.results.txt`
So maybe `centrifuge-kreport -x \$db_name ${prefix}.results.txt > ${prefix}.results.txt` ?

Reply:
Oops, yes confused myself -  updated the sample.

If output of `centrifuge_centrifuge` is _already_ `<prefix>.results.txt`, te nyou can just specify the variable alone (don't need to reconstruct it with `${prefix}`

Your example above wouldn't work as that would _definitely_ lead to a file name collision (both input and output would be the same), but if we don't specify `kreport`, and just `${prefix}.txt`, this is OK as this doesn't have `results` by default ,as we are creating the file ourselvs.

Reply:
Sounds good!